cmake/flash/CMakeLists.txt

```diff
@@ -1,20 +1,64 @@
 # SPDX-License-Identifier: Apache-2.0
 
-add_custom_target(runner_yml_props_target)
-
-function(runner_yml_write content)
+function(runners_yaml_append content)
   # Append ${content}\n to a target property which is later evaluated as a
   # generator expression when writing the flash runner yaml file.
   # We define this function here to have access to the `flash` target.
 
   set_property(
-    TARGET         runner_yml_props_target
+    TARGET         runners_yaml_props_target
     APPEND_STRING
     PROPERTY       yaml_contents
     "${content}\n"
     )
 endfunction()
 
+function(get_runners_prop prop out_var)
+  # Get property 'prop' from runners_yaml_props_target. An optional
+  # third argument can be used to provide a default value if the value
+  # is not found (set to ...-NOTFOUND).
+
+  get_target_property(out runners_yaml_props_target "${prop}")
+
+  if(("${ARGC}" EQUAL 3) AND ("${out}" STREQUAL "out-NOTFOUND"))
+    set("${out_var}" "${ARGV2}" PARENT_SCOPE)
+  else()
+    set("${out_var}" "${out}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+function(runners_yaml_append_config)
+  # Append the common configuration values to the relevant property target.
+
+  runners_yaml_append("\n# Common runner configuration values.")
+  runners_yaml_append("config:")
+  runners_yaml_append("  board_dir: ${BOARD_DIR}")
+  get_runners_prop(elf_file elf "${KERNEL_ELF_NAME}")
```

Comment:
because of the quotes. then `"${KERNEL_ELF_NAME}"` this will always become an argument to `get_runners_prop()`.
If `KERNEL_ELF_NAME` is not defined, then the argument becomes `""`, as in:
```
get_runners_prop(elf_file elf "")
```
alas an empty string.

So why not make the third argument mandatory and get rid of all the `ARGC` handling.

Reply:
and same goes for all other calls to: `get_runners_prop`